frontend/src/global_styles/content/_toast.sass

```diff
@@ -145,31 +149,36 @@ $nm-upload-box-padding: rem-calc(15) rem-calc(25)
   &.-error
     @extend %error-placeholder
 
-    &::before
+    &:not(.toast-icon)::before
       @include messages-icon
       @extend %nm-icon-error
 
   &.-success
     @extend %success-placeholder
 
-    &::before
+    &:not(.toast-icon)::before
       @include messages-icon
       @extend %nm-icon-success
 
   &.-warning
     @extend %warning-placeholder
 
-    &::before
+    &:not(.toast-icon)::before
       @include messages-icon
       @extend %nm-icon-warning
 
   &.-info
     @extend %info-placeholder
 
-    &::before
+    &:not(.toast-icon)::before
       @include messages-icon
       @extend %nm-icon-info
 
+  &.-notification-icon
+    &::before
+      @include messages-icon
+      @extend %nm-icon-notification
+
```

Comment:
I think you are over complicating things here. By creating a specific `-{{icon}}-icon` class, you have to add a CSS rule for **every** possible icon that you want to pass. Further, the modifier for an icon should only change the icon itself and not the whole toast. Why don't you simply use the classes we already have to render icons?`The `-info` class is still set on the toast, so there is no need for the syling if you adapt the rules above so that `toast-icon` also only affects the icon and not the styling.